### PR TITLE
Fix regression where many blank lines were reported in browse mode

### LINKED ISSUE #16554: Many blank lines in browse mode after #16471 merge
### Steps to reproduce:
1. Open https://github.com/nvaccess/nvda/pull/16471
2. Go to the `Code Review Checklist:` heading
3. Press down arrow

### Actual behavior:
NVDA reports three empty lines below the heading before it reports the first list item

### Expected behavior:
NVDA directly moves to the first list item.

### Additional context
There are also empty lines added in the github pull request overview, for example.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-31919,383b2044

#### Windows version:
Windows 11 23H2

#### Name and version of other software in use when reproducing the issue:
Firefox 125.0.3

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, I reverted a1848dd when running from source and the issue disappeared.

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A

### PR DESCRIPTION
**Pull request by SaschaCowley**
### Link to issue number:

Fixes #16554

### Summary of the issue:

#16471  introduced a regression whereby many blank lines were reported in browse mode.

### Description of user facing changes

Extraneous blank lines are no longer reported in browse mode.

### Description of development approach

Updated checks in `gecko_ia2.cpp`:

- Check that the length of `name` is non-zero (`SysStringLen` returns `0` if the `BSTR` passed is `null`)
- Added check that `description`'s `value` is not the empty string, as checking that it has a value is necessary but not sufficient.

### Testing strategy:

Tested reading the PR for #16471 as described in #16554. Also tested as described in #16471  to ensure changes did not break the new behaviour.

### Known issues with pull request:

N/A

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by cary-rowen** [comment]
Hi,
Shouldn't regression fixes be for beta versions?

**Comment by seanbudd** [comment]
@cary-rowen - this regression is on alpha, caused by a PR targeting 2024.3. It's not in the 2024.2 code.

@SaschaCowley - I assume you are still working on this as the diff appears to be empty, so I've marked this a draft

**Comment by cary-rowen** [comment]
Sorry, I misremembered this

**Review by LeonarddeR — APPROVED** [review]
Thanks for fixing it!

File: nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp [hunk 1]
```diff
@@ -1286,7 +1286,7 @@ VBufStorage_fieldNode_t* GeckoVBufBackend_t::fillVBuf(
 			parentNode->isBlock=false;
 		}
 
-		if ((isInteractive || role == ROLE_SYSTEM_SEPARATOR || name || description.has_value()) && parentNode->getLength() == 0) {
+		if ((isInteractive || role == ROLE_SYSTEM_SEPARATOR || SysStringLen(name) > 0 || (description.has_value() && !description.value().empty())) && parentNode->getLength() == 0) {
```
Comment (seanbudd) [inline nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp:1289]: This is challenging to parse given the line length and complexity

```suggestion
		if (
			(
				isInteractive
				|| role == ROLE_SYSTEM_SEPARATOR
				|| SysStringLen(name) > 0
				|| (
					description.has_value()
					&& !description.value().empty()
				)
			)
			&& parentNode->getLength() == 0
		) {
```

**Review by seanbudd — APPROVED** [review]

**Comment by Adriani90** [comment]
It seems there are still blank lines added from time to time to the virtual document when web pages update content dynamically.E.g. on Github insights page while the time stamps for the issues / PRs are updating.

**Comment by Adriani90** [comment]
But they seem to disappear when pressing nvda+f5 to update the virtual document manually.

**Comment by SaschaCowley** [comment]
Can you open a new issue for this, and if possible provide a concrete repro case?

**Comment by Adriani90** [comment]
I can try to report a concrete case and isolate it as far as possible, it seems it happens only in Firefox.

**Pull request merged**